src/kademlia/dht_storage.cpp

```diff
@@ -568,10 +572,17 @@ namespace
 	};
 }
 
-dht_storage_interface* dht_default_storage_constructor(sha1_hash const& id
-	, dht_settings const& settings)
+void dht_storage_counters::reset()
+{
+	torrents = 0;
+	peers = 0;
+	immutable_data = 0;
+	mutable_data = 0;
+}
+
+dht_storage_interface* dht_default_storage_constructor(dht_settings const& settings)
```

Comment:
not for this patch, but at some point this function should return a unique_ptr<> to the storage, and it should probably be held by a unique_ptr<> in the session too